chrome/browser/android/adblock/regex_resource_type_detector.cpp

```diff
@@ -0,0 +1,54 @@
+/*
+ * This file is part of Adblock Plus <https://adblockplus.org/>,
+ * Copyright (C) 2006-present eyeo GmbH
+ *
+ * Adblock Plus is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * Adblock Plus is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Adblock Plus.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "chrome/browser/android/adblock/regex_resource_type_detector.h"
+#include <regex>
+
+namespace
+{
+    const std::regex RE_JS("\\.js(?:\\?.+)*$", std::regex_constants::ECMAScript | std::regex_constants::icase);
+    const std::regex RE_CSS("\\.css$", std::regex_constants::ECMAScript | std::regex_constants::icase);
+    const std::regex RE_IMAGE("\\.(?:gif|png|jpe?g|bmp|ico)$", std::regex_constants::ECMAScript | std::regex_constants::icase);
+    const std::regex RE_FONT("\\.(?:ttf|woff)$", std::regex_constants::ECMAScript | std::regex_constants::icase);
+    const std::regex RE_HTML("\\.html?$", std::regex_constants::ECMAScript | std::regex_constants::icase);
```

Comment:
Just wonder, whether `(?:\\?.+)*` should be used with all regexps or not. It's a common trick to add some query parameters to bypass caching and sometimes to collect some metrics.

Reply:
yes, me too. We need somebody else to join discussion

Reply:
What about this question?

@shoniko @hfiguiere